services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentService.kt

```diff
@@ -0,0 +1,84 @@
+
+import fi.hel.haitaton.hanke.application.ApplicationEntity
+import fi.hel.haitaton.hanke.application.ApplicationNotFoundException
+import fi.hel.haitaton.hanke.application.ApplicationRepository
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentMetadata
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentType
+import fi.hel.haitaton.hanke.attachment.common.AttachmentNotFoundException
+import fi.hel.haitaton.hanke.attachment.common.AttachmentScanStatus.OK
+import fi.hel.haitaton.hanke.attachment.common.AttachmentValidator.validate
+import fi.hel.haitaton.hanke.currentUserId
+import java.time.OffsetDateTime.now
+import java.util.UUID
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.multipart.MultipartFile
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class ApplicationAttachmentService(
+    private val applicationRepository: ApplicationRepository,
+    private val attachmentRepository: ApplicationAttachmentRepository
+) {
+    @Transactional(readOnly = true)
+    fun getMetadataList(applicationId: Long): List<ApplicationAttachmentMetadata> =
+        findApplication(applicationId).attachments.map { it.toMetadata() }
+
+    @Transactional(readOnly = true)
+    fun getContent(applicationId: Long, attachmentId: UUID): Pair<String, ByteArray> {
+        val attachment = findApplication(applicationId).attachments.findBy(attachmentId)
+
+        if (attachment.scanStatus != OK) {
+            throw AttachmentNotFoundException()
+        }
+
+        return Pair(attachment.fileName, attachment.content)
+    }
+
+    @Transactional
+    fun addAttachment(
+        applicationId: Long,
+        attachmentType: ApplicationAttachmentType,
+        attachment: MultipartFile
+    ): ApplicationAttachmentMetadata {
+        val application = findApplication(applicationId)
+        val file = validate(attachment)
+        val applicationAttachment =
+            ApplicationAttachmentEntity(
+                id = null,
+                fileName = file.originalFilename!!,
+                content = file.bytes,
+                createdByUserId = currentUserId(),
+                createdAt = now(),
+                scanStatus = OK,
+                attachmentType = attachmentType,
+                application = application,
+            )
+
+        return attachmentRepository.save(applicationAttachment).toMetadata().also {
+            logger.info { "Added attachment ${it.id} to application $applicationId" }
+        }
+    }
+
+    @Transactional
+    fun deleteAttachment(applicationId: Long, attachmentId: UUID) {
+        val attachment = findApplication(applicationId).attachments.findBy(attachmentId)
```

Comment:
We should probably refuse deleting attachments after the application has moved to HANDLING.

Reply:
This was discussed and decided that the deletion / modification restrictions will be implemented in the upcoming task: https://helsinkisolutionoffice.atlassian.net/browse/HAI-1545